### DISCUSSION TITLE
Multiple background profiles with associated alarm tables

### DISCUSSION THREAD

**Discussion by Backblocks** [artisan-roaster-scope/artisan#1451] [General]
I am trying to set up Artisan so that we have a distinct background curve for each bean we roast, and along with the curve there is to be a unique alarm table.  

I can set up a button to pull in the background curve for controlling the PID for controlling the temperature when roasting a particular bean, but have been unable to get an alarm table to come along with it.  And cant figure out how to save the customised alarm table along with the background curve.  I can get the table to come back but then the curve has disappeared.  Probably a simple answer but it is driving me nuts and I am arguably too close to the edge anyway.  

So I guess the short question is: How do I save the alarm table along with the background curve for controlling the PID in such a way that the curve and the unique alarm table both come back when needed?

Thanks heaps for any light shed.  :-)

**Comment by MAKOMO** [comment]
Why not just tick "Load alarms from background" in the Alarms dialog (menu `Config >> Alarms`)? The current alarm table is save within the *.alog file on saving a profile.

Not sure what you refer to by "then the curve has disappeared". Maybe some more words on your exact setup, software and hardware version and Artisan configuration could help. It may also help to post just one single questions per thread and have your questions answered one by one.

Note that the alarm mechanism is completely independent from the PID controller.

<img width="800" alt="Screenshot 2024-02-15 at 10 30 51" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/c7ced47c-9a5e-4ce0-8e39-ed906c336a5f">

**Reply by Backblocks** [reply]
Thanks for the reply.  

I did try the 'load from background' earlier but could not achieve what I needed.  What I kept finding is that when I saved the .alog file, the alarm table was preserved, but the Profile Background (the curve that disappears as I put it) is not preserved.  However, in attempting to find a solution yesterday, I managed to get find the answer.  The problem was I had the wrong starting point.

I was opening the Profile Background I wanted to use from the Roast>Background dialog box in Artisan, altering the Alarm Table, and then saving it.  But as I said the Background Profile is gone and only the Alarm Table saved when it is loaded up next time.

What I discovered is that I needed to open the Profile Background .alog file from File Explorer, make the Alarm Table changes, and then it saves successfully preserving the Profile Background and the Alarm Table when you load it next time.  But I guess that you are more than well aware of this.  So I am all sorted now thank you.

You asked about my setup.  Artisan 2.8.4 with Windows 11.  We have an older 10 kg Toper drum roaster, with a few Phidgets and some other engineering bits and pieces added, and along with a bit of imagination we are on the road to fully automated with Artisan.  Almost there . . .

As a final thought, I would like to say that I love Artisan and what it is able to do.  Once you get your head around how it works you can make things work rather nicely.  You and the team have done a great job.  Thanks again.
